### PR TITLE
Added implementation for Redis SCAN command

### PR DESCRIPTION
**Pull request by SSPkrolik**
Added implementation for Redis SCAN command overloaded for different set of parameters.

### CODE REVIEW

**Comment by Araq** [comment]
@dom96 ping.

File: lib/pure/redis.nim [hunk 1]
```diff
@@ -285,6 +285,30 @@ proc keys*(r: Redis, pattern: string): RedisList =
   r.sendCommand("KEYS", pattern)
   return r.readArray()
 
+proc scan*(r: Redis, cursor: var BiggestInt): RedisList =
```
Comment (dom96) [inline lib/pure/redis.nim]: Why must ``cursor`` be a var param?

**Comment by SSPkrolik** [comment]
@dom96 Because Redis answers with a bulk reply containing keys, and a new value for cursor, so I thought making var cursor would be a nice solution for API design.

**Comment by SSPkrolik** [comment]
@dom96 Could you provide any feedback on the code so I could continue working on Redis driver?

**Comment by dom96** [comment]
That's ok. Looking towards the future I think we want an asynchronous redis driver and at that point var params will need to be avoided. Would be awesome if you could develop that :)

**Pull request merged**